core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -223,7 +223,18 @@ default Optional<ConnectorPartitioningHandle> getCommonPartitioningHandle(Connec
      *
      * @throws RuntimeException if table handle is no longer valid
      */
-    @Deprecated // ... and optimized implementations already removed
+    default SchemaTableName getTableName(ConnectorSession session, ConnectorTableHandle table)
```

Comment:
do we need to do this refactor (rename)? Maybe separate commit?

Reply:
i did this because this method was deprecated and almost removed, i didn't want ot add new usages for a method that's already known not to be implemented
would it help if we do https://github.com/trinodb/trino/pull/17267 first?